src/researches/metaDescriptionKeyword.js

```diff
@@ -1,35 +1,87 @@
+import matchWords from "../stringProcessing/matchTextWithArray";
 import getSentences from "../stringProcessing/getSentences";
-import { findTopicFormsInString } from "./findKeywordFormsInString";
 
 /**
- * Matches the keyword in the description if a description and keyword are available.
- * Returns null if no description is specified in the given paper.
+ * Replaces found keyword forms in the given description.
+ *
+ * @param {string} description the description to remove the matched keyword forms from.
+ * @param {Object[]} matchedKeywordForms the matched keyword forms to remove from the description.
+ * @returns {string} the description with the keywords removed.
+ */
+const replaceFoundKeywordForms = function( description, matchedKeywordForms ) {
```

Comment:
From what I understood, this function replaces all found matches of all found words, regardless of whether a full match was found or not. This means that in case:
```
keyphrase: "vacation in Spain",
synonyms: "vacation in Iberia",
description: "An awesome meta description about vacations in Iberia."
```
"vacation" will already be stripped after keyphrase search and in the end neither synonyms nor keywords would be matched.
Is that right?